src/vecs/client.py

```diff
@@ -96,12 +96,33 @@ def get_collection(self, name: str) -> Collection:
             CollectionNotFound: If no collection with the given name exists.
         """
         from vecs.collection import Collection
-
-        collections = Collection._list_collections(self)
-        for collection in collections:
-            if collection.name == name:
-                return collection
-        raise CollectionNotFound("No collection found with requested name")
+        query = text(
+        f"""
+        select
+            relname as table_name,
+            atttypmod as embedding_dim
+        from
+            pg_class pc
+            join pg_attribute pa
+                on pc.oid = pa.attrelid
+        where
+            pc.relnamespace = 'vecs'::regnamespace
+            and pc.relkind = 'r'
+            and pa.attname = 'vec'
+            and not pc.relname ^@ '_'
+            and pc.relname = '{name}'
+        """
+        )
+        collections = []
+        with self.Session() as sess:
+            for name, dimension in sess.execute(query):
+                existing_collection = Collection(name, dimension, self)
+                collections.append(existing_collection)
```

Comment:
since we can only ever get 1 or 0 matches here, how about removing the `collections` variable, and the for loop in favor of something like

```python
# type signature shown just for illustration, no need to include it
query_result: Row[Tuple[str, int]] | None = sess.execute(query).fetchone()
if query_result is None:
    raise ...
name, dimension = query_result

return Collection(name, dimension, self)
```